### DISCUSSION TITLE
issue: There's no focus on error field when disabled field with the same name exists

### DISCUSSION THREAD

**Discussion by sergeyivanovv89** [react-hook-form/react-hook-form#9976] [Q&A] [answered]
### Version Number

7.43.1

### Codesandbox/Expo snack

https://codesandbox.io/s/rhf-bug-with-no-focus-when-disabled-input-exists-cww1z2

### Steps to reproduce

1. Click on 'Submit' button.
2. See error.

### Expected behaviour

The first input should be focused.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment, marked as answer]
If you remove the `disabled` prop from the second `input1` input, you'll see that the second input registration overrides the `input1` properties _(RHF focuses the second `input1` input on error)_.

You can register 2 different inputs with the same name with RHF.

As a side note, avoid defining React components inside other components to avoid re-initialization on each parent component render and potentially `ref` issues _(input's `ref` will be different on each render)_

**Reply by sergeyivanovv89** [reply]
I've just updated my codesandbox moving `InputField` into a separate file and it didn't take effect.
